tokio-process/src/unix/mod.rs

```diff
@@ -126,49 +138,55 @@ impl Kill for Child {
 }
 
 impl Future for Child {
-    type Item = ExitStatus;
-    type Error = io::Error;
+    type Output = io::Result<ExitStatus>;
 
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        self.inner.poll()
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+        (&mut Pin::get_mut(self).inner).poll_unpin(cx)
     }
 }
 
 #[derive(Debug)]
-pub struct Fd<T>(T);
+pub struct Fd<T>
+where
+    T: AsRawFd + Unpin,
```

Comment:
Given that `Fd` doesn't have a `Drop` impl which requires any of these bounds, we can simplify things by omitting them on the definition (so we don't have to repeat all the bounds everywhere)